definitions/scenarios/upgrade_to_foreman_3_3.rb

```diff
@@ -0,0 +1,91 @@
+module Scenarios::Foreman_3_3
+  class Abstract < ForemanMaintain::Scenario
+    def self.upgrade_metadata(&block)
+      metadata do
+        tags :upgrade_scenario
+        confine do
+          !feature(:satellite) && feature(:foreman)
+          (feature(:foreman_server).current_minor_version == '3.2' || \
+          ForemanMaintain.upgrade_in_progress == '3.3')
+        end
+        instance_eval(&block)
+      end
+    end
+
+    def target_version
+      '3.3'
+    end
+  end
+
+  class PreUpgradeCheck < Abstract
+    upgrade_metadata do
+      description 'Checks before upgrading to Foreman 3.3'
+      tags :pre_upgrade_checks
+      run_strategy :fail_slow
+    end
+
+    def compose
+      add_steps(find_checks(:default))
+      add_steps(find_checks(:pre_upgrade))
+    end
+  end
+
+  class PreMigrations < Abstract
+    upgrade_metadata do
+      description 'Procedures before migrating to Foreman 3.3'
+      tags :pre_migrations
+    end
+
+    def compose
+      add_steps(find_procedures(:pre_migrations))
+    end
+  end
+
+  class Migrations < Abstract
+    upgrade_metadata do
+      description 'Migration scripts to Foreman 3.3'
+      tags :migrations
+      run_strategy :fail_fast
+    end
+
+    def set_context_mapping
+      context.map(:assumeyes, Procedures::Installer::Upgrade => :assumeyes)
+    end
+
+    def compose
+      add_step(Procedures::Repositories::Setup.new(:version => '3.3'))
+      modules_to_enable = ['ruby:2.7', 'foreman:el8']
```

Comment:
The foreman module by itself should be enough here, it has the dependency on the Ruby module.

Reply:
removed